### PR TITLE
Broadcast on leader changing

### PR DESCRIPTION
**Pull request by markcty**

### CODE REVIEW

File: curp/src/server/state.rs [hunk 1]
```diff
@@ -50,6 +50,8 @@ pub(super) struct State<C: Command + 'static> {
     pub(super) cmd_board: Arc<Mutex<CommandBoard>>,
     /// Last time a rpc is received.
     pub(super) last_rpc_time: Arc<RwLock<Instant>>,
+    /// Leader changes tx
+    leader_tx: broadcast::Sender<ServerId>,
```
Comment (themanforfree) [inline curp/src/server/state.rs:54]: the server didn't send the `ServerId` of new leader

File: curp/src/server/mod.rs [hunk 2]
```diff
@@ -557,9 +564,7 @@ impl<C: 'static + Command> Protocol<C> {
         if req.term > state.term {
             state.update_to_term(req.term);
```
Comment (themanforfree) [inline curp/src/server/mod.rs:565]: `update_to_term` will  change `leader_id` to `None`, other places where called this function may not immediately update the `leader_id`, e.g.: [link](https://github.com/themanforfree/Xline/blob/780f5bbc537d786b71cfe0e143448938cf49adea/curp/src/server/bg_tasks.rs#L195), so there will be a period of time we don't know who the leader is, should we notify this case to xline?

Comment (markcty) [reply curp/src/server/mod.rs:565]: Fixed

File: curp/src/server/mod.rs [hunk 3]
```diff
@@ -189,6 +189,13 @@ impl<C: Command + 'static> Rpc<C> {
             .await?;
         Ok(())
     }
+
+    /// Get a subscriber for leader changes
+    #[inline]
+    #[must_use]
+    pub fn leader_rx(&self) -> broadcast::Receiver<Option<ServerId>> {
```
Comment (rogercloud) [inline curp/src/server/mod.rs:196]: This subscriber is never used in this change. Why?

**Review by themanforfree — APPROVED** [review]

File: curp/src/server/mod.rs (hunk 3 above)
Comment (markcty) [reply curp/src/server/mod.rs:196]: I guess @themanforfree will add the usage in his fork in the next PR.

Comment (themanforfree) [reply curp/src/server/mod.rs:196]: I think we can add tests for this, like you did for client in this PR

Comment (markcty) [reply curp/src/server/mod.rs:196]: Sure. Added.

**Review by themanforfree — APPROVED** [review]

**Review by rogercloud — APPROVED** [review]

**Pull request merged**